File: CppSolution/LongestCommonPrefix.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) 
    {
    	string r;
    	if (strs.size() < 1) return r;
        for (int i = 0; i < strs[0].size(); i++)
        {
        	if (strs.size() == 1) return strs[0];
        	for (int j = 1; j < strs.size(); j++)
        	{
        		if (i == strs[j].size()) return r;
        		if (strs[0][i] != strs[j][i])
        		{
        			return r;
        		}
        	}
        	r += strs[0][i];
        }
        return r;
    }
};
int main()
{
	return 0;
}

File: rust_solution/src/longest_common_prefix.rs
/*
    Write a function to find the longest common prefix string amongst an array of strings.

    If there is no common prefix, return an empty string "".
*/
pub struct Solution {}

impl Solution {
    pub fn longest_common_prefix(strs: &Vec<String>) -> String {
        let mut result = String::from("");
        let mut i = 0;
        if strs.len() < 1 || strs[0].len() < 1 {
            return result;
        }

        let mut c;
        loop {
            c = strs[0].as_bytes()[i] as char;
            for s in strs {
                if i == s.len() || s.as_bytes()[i] as char != c {
                    return result;
                } 
            }
            result.push(c);
            i += 1;
        }
    }
}